src/main/java/pt/tqs/g205/services/CategoriaPratoService.java

```diff
@@ -0,0 +1,30 @@
+package pt.tqs.g205.services;
+
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import pt.tqs.g205.domain.CategoriaPrato;
+import pt.tqs.g205.repositories.CategoriaPratoRepository;
+
+@Service
+public class CategoriaPratoService {
+
+  @Autowired
+  private CategoriaPratoRepository categoriaRepo;
+
+  public List<CategoriaPrato> getAll() {
+    List<CategoriaPrato> categorias = categoriaRepo.findAll();
+    categorias.forEach(e -> e.setPratos(null));
+    return categorias;
+  }
+
+  public CategoriaPrato getById(Integer id) {
+    Optional<CategoriaPrato> cat = categoriaRepo.findById(id);
+
+    return cat.orElseThrow(() -> new NoSuchElementException());
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Replace this lambda with a method reference. ([squid:S1612](https://sonarcloud.io/organizations/bdomingues-github/rules#rule_key=squid%3AS1612))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=pt.tqs.g205%3Arestapi&pullRequest=5&issues=AWNCwXmSXOBV_Yrh-xNw&open=AWNCwXmSXOBV_Yrh-xNw)

[ISSUE_KEY]: AWNCwXmSXOBV_Yrh-xNw